growth/too/models.py

```diff
@@ -1007,3 +1012,247 @@ class Candidate(db.Model):
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object mag')
+
+    ssdistnr = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object distance')
+
+    CLU_sfr_ha = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate Halpha')
+
+    CLU_mstar = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU stellar mass')
+
+    CLU_sfr_fuv = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate fuv')
+
+    CLU_z = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU redshift')
+
+    CLU_name = db.Column(
+        db.String,
+        nullable=True,
+        comment='CLU name')
+
+    CLU_id = db.Column(
+        db.Integer,
+        nullable=True,
+        comment='CLU ID')
+
+    CLU_d_to_galaxy_arcsec = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU distance arcsec')
+
+    comment = db.Column(
+        db.String,
+        nullable=True,
+        comment='Comment')
+
+    lightcurve = db.relationship(lambda: Lightcurve)
+
+
+class Lightcurve(db.Model):
+    """Candidate light curve pulled from the GROWTH
+    Marshal"""
+
+    name = db.Column(
+        db.ForeignKey(Candidate.name),
+        primary_key=True,
+        nullable=False,
+        comment='Candidate name')
+
+    date_observation = db.Column(
+        db.ARRAY(db.DateTime),
+        nullable=True,
+        comment='Observation date')
+
+    fil = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Filter')
+
+    instrument = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Instruments')
+
+    limmag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Limiting magnitude')
+
+    mag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag PSF')
+
+    magerr = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag uncertainty')
+
+    exptime = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Exposure time')
+
+    programid = db.Column(
+        db.ARRAY(db.Integer),
+        nullable=True,
+        comment='Program ID number (1,2,3)')
+
+    first_detection_time_tmp = db.Column(
+        db.DateTime,
+        nullable=True,
+        comment='Time first detection')
+
+    @hybrid_property
+    def first_detection_time(self):
+        if self.date_observation is None:
+            return None
+        jd_array = time.Time(self.date_observation,
+                             format='datetime').jd
+        min_jd = np.min(jd_array)
+        return time.Time(min_jd, format='jd').datetime
+
+    @first_detection_time.expression
+    def first_detection_time(cls):
+        return func.min(cls.date_observation)
+
+    @property
+    def first_detection_mag(self):
```

Comment:
OK, these can all be implemented faster in SQL. No need to implement them in Python like this.

Reply:
We could have this fix as a separate pull request.

Reply:
Please. This is really inefficient.

Reply:
And this is really easy to fix.

Reply:
After the sqlalchemy query, where I get my candidates and filter them (with both sqlalchemy and python), I have the following loop:

for source, coords in zip(sources_growth_marshal_contour, sources_coords):
        source["ra"], source["dec"] = coords.ra, coords.dec
        date_list = list(s.date_observation for s in source['lightcurve'])
        source["first_detection_time"] = min(date_list)
        source["latest_detection_time"] =  max(date_list)
        index_max = date_list.index(max(date_list))
        source["latest_detection_mag"] = list(s.mag for s in source['lightcurve'])[index_max]
        source["latest_detection_magerr"] = list(s.magerr for s in source['lightcurve'][index_max]

Should I convert these operations into new sqlalchemy queries?  Note that this is usually done on a handful of candidates, say few tens.